users/views.py

```diff
@@ -154,6 +155,15 @@ def user_create_confirm(request, token):
         user.confirm_token = ''
         user.is_active = True
         user.save()
+        user.circle_set.create(
+            name='Me',
+            is_editable=False
+        )
+        user.circle_set.create(
+            name='Public',
+            is_editable=False
+        )
```

Comment:
Handling the two edge cases will be more complicated as having a big overhead. I think the best approach is the combination of both, i.e. handling the public circle as non value and creating one circle for every user for the 'Me' case. 
Any other possible solution will take much more time to implement, what due our time issues I wouldn't waste so much time on that